### ISSUE TITLE
False-positive `unintentional comparison` error with closures

### ISSUE THREAD

**Issue by e-kucheriavyi** [microsoft/TypeScript#64314]
### 🔎 Search Terms

unintentinal, comparison, 2367, overlap

### 🕗 Version & Regression Information

- typescript: `v7.0.2`
- node: `v24.14.1`
- pnpm: `v12.3.4`

### 💻 Code

```ts
let n: "a" | "b" = "a";

const updateN = () => {
	n = "b";
};

// This comparison appears to be unintentional because the types '"a"' and '"b"' have no overlap. [2367]
if (n === "b") {}

updateN();

// This comparison appears to be unintentional because the types '"a"' and '"b"' have no overlap. [2367]
if (n === "b") {}

n = "b";

// No error
if (n === "b") {}
```

### 🙁 Actual behavior

If a variable of type `"a" | "b"` is changed in a function with closure, the LSP fails to catch the change, and assumes that it was never changed.

### 🙂 Expected behavior

No error warning, since code works as expected

**Comment by MartinJohns** [comment]
Duplicate of #9998.
